### PR TITLE
Parametrizing deploy timeout

### PR DESCRIPTION
**Pull request by jpodivin**
Rather simple adjustment allowing us to use different deployment timeouts should we wish. 
The existing 15 sec is still in place, but now we can choose another constant, for example if we are dealing with constrained env.

### CODE REVIEW

**Comment by danpawlik** [comment]
recheck

**Comment by jpodivin** [comment]
recheck
```
Failed to watch *unstructured.Unstructured: unknown\nI1214 10:45:09.336338   33204 trace.go:236] Trace[65370265]: \"Reflector ListAndWatch\" name:k8s.io/client-go/tools/watch/informerwatcher.go:146 (14-Dec-2023 10:44:56.688) (total time: 12647ms):\nTrace[65370265]: ---\"Objects listed\" error:<nil> 12647ms (10:45:09.336)\nTrace[65370265]: [12.647576916s] [12.647576916s] END\nE1214 10:55:17.083078   33204 reflector.go:148] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *unstructured.Unstructured: the server has asked for the client to provide credentials\nE1214 11:02:37.572829   33204 reflector.go:148] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *unstructured.Unstructured: the server has asked for the client to provide credentials
```

**Review by fao89 — APPROVED** [review]
minor comment, but LGTM

File: api/v1beta1/openstackdataplanedeployment_types.go [hunk 1]
```diff
@@ -45,6 +45,12 @@ type OpenStackDataPlaneDeploymentSpec struct {
 	// +kubebuilder:validation:Optional
 	// ServicesOverride list
 	ServicesOverride []string `json:"servicesOverride"`
+
+	// Time before the deployment is requeued in seconds
+	// +kubebuilder:validation:Optional
```
[the remark this answers is not in the document]
Comment (jpodivin) [reply api/v1beta1/openstackdataplanedeployment_types.go:50]: Good point.

**Review by fao89 — APPROVED** [review]

**Comment by jpodivin** [comment]
recheck
`{"attempts": 60, "changed": true, "cmd": ["oc", "get", "po", "-l", "osp-provisionserver/name=openstack-edpm-ipam-provisionserver", "-n", "openstack", "-o", "name"], "delta": "0:00:00.125222", "end": "2024-01-09 10:13:04.630265", "msg": "", "rc": 0, "start": "2024-01-09 10:13:04.505043", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}`

**Pull request merged**